pyclient/apps/rule.py

```
@@ -0,0 +1,207 @@
+# If rules not present in policy, program adds first rule to policy.
+def addRule(rule, policy, name):
+    if (hasattr(policy.spec, "rules")):
+        policy.spec.rules.append(rule)
+        api_instance.update_network_security_policy("default", name, policy)
+        return ("Rule appended to policy!")
+    else:
+        policy = SecurityNetworkSecurityPolicy(meta=ApiObjectMeta(name=name),
+                                               spec=SecurityNetworkSecurityPolicySpec(rules=[rule], attach_tenant=True))
+        api_instance.update_network_security_policy("default", name, policy)
+        return ("First rule added in policy.")
+
+# Deletes rule by using previous rules and removing the rule which is to be deleted. The policy is then updated
+def deleteRule(rule, policy, name):
+    exists = False
+    if (hasattr(policy.spec, "rules")):
+        exists = rule in policy.spec.rules
+
+    if (exists == True):
+        policy.spec.rules.remove(rule)
+        api_instance.update_network_security_policy("default", name, policy)
+        return("Rule deleted!")
+    else:
+        return("Rule doesn't exist. Double check your inputs!")
+
+# Updates rule by finding old rule and replacing it with new rule. Policy is then updated
+def updateRule(rule, newRule, policy, name):
+    exists = False
+    if (hasattr(policy.spec, "rules")):
+        exists = rule in policy.spec.rules
+
+    if (exists == True):
+        newRules = policy.spec.rules
+        newRules[newRules.index(rule)] = newRule
+        newPolicy = SecurityNetworkSecurityPolicy(meta=ApiObjectMeta(name=name),
+                                               spec=SecurityNetworkSecurityPolicySpec(rules=newRules, attach_tenant=True))
+        api_instance.update_network_security_policy("default", name, newPolicy)
+        print("Rule updated!")
+    else:
+        print("Rule doesn't exist. Double check your inputs!")
+
+def handleErrorResponse(response):
```

it would make sense to make this a common utility routine instead keeping this here. A routine that returns string for the error string for http return codes.

---

Also, please use logging.error to raise exceptions.
You can also use response.raise_for_status() instead of checking each error code and returning an error.
https://docs.python-requests.org/en/master/api/#requests.Response.raise_for_status